src/datasets/utils/metadata.py

```diff
@@ -173,14 +194,42 @@ def to_builder_configs(self, builder_config_cls):
                     You can resolve data files with either .resolve_data_files_locally() or
                     .resolve_data_files_in_dataset_repository() method."""
                 )
+        ignored_params = [
+            param
+            for meta_config in metadata_configs.values()
+            for param in meta_config
+            if hasattr(builder_config_cls, param) and param != "default"
```

Comment:
```suggestion
            if not hasattr(builder_config_cls, param) or param == "default"
```
if I understood it right, this should be a negation of what is in return statement.

Reply:
good catch, I think it should be 
```python
if not hasattr(builder_config_cls, param) and param != "default"
```

Reply:
ah true